src/screens/Forum/EditForumPostScreen.js

```diff
@@ -0,0 +1,213 @@
+import React, {
+  useState, useEffect, useCallback, useRef,
+} from 'react';
+import {
+  ScrollView, Picker, Alert, StyleSheet, View,
+} from 'react-native';
+import {
+  TextInput, Button, Text,
+} from 'react-native-paper';
+// import auth from '@react-native-firebase/auth';
+import firestore from '@react-native-firebase/firestore';
+import PropTypes from 'prop-types';
+import { theme } from '../../style';
+
+export default function EditForumPostScreen({ navigation }) {
+  const [title, setTitle] = useState('');
+  const [body, setBody] = useState('');
+  const [errorMessageTitle, setErrorMessageTitle] = useState(null);
+  const [errorMessageBody, setErrorMessageBody] = useState(null);
+  const [errorMessageFirestore, setErrorMessageFirestore] = useState(null);
+
+  const postID = navigation.getParam('postID');
+
+  const [categoryID, setCategoryID] = useState('');
+
+  /* TO DO: Mount Categories ID */
+  /*
+    const [forumCategories, setForumCategories] = useState([]);
+    firestore().collection('forum_categories')
+        .get()
+        .then((snapshot) => {
+          const forumCategories = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
+          this.setState({ forumCategories });
+        });
+      this.unsubscribe = auth().onAuthStateChanged((user) => {
+        if (user) {
+          this.setState({ userID: user.uid });
+        }
+      });
+  */
+
+  const getData = useCallback(async () => {
+    // setLoading(true);
+
+    /* Query forum post data from postid passed in through navigation */
+    const postSnap = await firestore().collection('forum_posts').doc(postID)
+      .get();
+    const postData = postSnap.data();
+
+    /* Get forum post details */
+    const postTitle = postData.title || 'Title unset';
+    const postBody = postData.body || 'Body unset';
+
+    setTitle(postTitle);
+    setBody(postBody);
+  }, [postID]);
+
+  /* Writes reply data to Firebase/Firestore */
+  const sendData = useCallback(async () => {
+    const postTitle = title;
+    const postBody = body;
+
+    if (title.length === 0) {
```

Comment:
These if statements could probably be better arranged (ie., if (title.length > 0 && body.length > 0) {} else {}